packages/jest-cli/src/jest.js

```diff
@@ -186,9 +186,19 @@ const runJest = (config, argv, pipe, onComplete) => {
           processor(runResults);
         }
         if (argv.json) {
-          process.stdout.write(
-            JSON.stringify(formatTestResults(runResults, config)),
-          );
+          if (argv.jsonOutputFile) {
+            fs.writeFileSync(
+              argv.jsonOutputFile,
+              JSON.stringify(formatTestResults(runResults, config)),
+            );
+            process.stdout.write(
+              `Wrote the test results in JSON to: ${argv.jsonOutputFile}\n`,
```

Comment:
Above do: `const outputFile = path.resolve(process.cwd(), argv.jsonOutputFile);`

and then here turn it back into a relative path:

`Test results written to: ${path.relative(process.cwd(), outputFile)}`

this way when passing an odd path like `../foo/../foo/test.json` it would print a normalized path like `../foo/test.json`.